src/cdomains/floatDomain.ml

```diff
@@ -36,11 +36,40 @@ module type FloatDomainBase = sig
   include FloatArith with type t := t
 
   val of_const : float -> t
+  val of_int: IntDomain.IntDomTuple.t -> t
+  val cast_to: Cil.ikind -> t -> IntDomain.IntDomTuple.t
 end
 module FloatInterval = struct
   type t = (float * float) option [@@deriving eq, ord, to_yojson]
 
 
+  let big_int_of_float f =
+    let x, n = Float.frexp f in
+    let shift = min 52 n in
+    let x' = x *. Float.pow 2. (Float.of_int shift) in
+    Big_int_Z.mult_big_int
+      (Big_int_Z.big_int_of_int64 (Int64.of_float x'))
+      (Big_int_Z.power_int_positive_int 2 (n - shift))
+
+  let cast_to ik = function
+    | None -> IntDomain.IntDomTuple.top_of ik
+    | Some (l, h) -> IntDomain.IntDomTuple.of_interval ik
+                       (* as converting from float to integer is (exactly) defined as leaving out the fractional part,
+                          (value is truncated towrad zero) we do not require specific rounding here *)
+                       (big_int_of_float l, big_int_of_float h)
```

Comment:
I personally would prefer not to have a comment in between arguments of a function application, but that's my taste and completely fine if you leave it like this :)